server/src/main/java/org/elasticsearch/index/mapper/BlockLoader.java

```diff
@@ -348,6 +349,11 @@ interface BlockFactory {
          */
         BytesRefBuilder bytesRefs(int expectedCount);
 
+        /**
+         * Build a builder to load {@link SpatialPoint}s backed by WKB in BytesRefBlock.
+         */
+        BytesRefBuilder geometries(int expectedCount);
```

Comment:
Could we remove this and use `bytesRefs`? If so, maybe we could reuse `BlockSourceLoader.BytesRefsBlockLoader`  like Ignacio was saying. It seems more likely.

Reply:
OK, I've removed that now, but could not remove GeometriesBlockLoader because it uses Geometries which does:

```
((BlockLoader.BytesRefBuilder) builder).appendBytesRef(new BytesRef(wkb));
```

While the equivalent one in the BytesRefsBlockLoader does:

```
((BlockLoader.BytesRefBuilder) builder).appendBytesRef(toBytesRef(scratch, (String) v));
```

So the BytesRefs one assuming incoming data is a String object, while the Geometries one assumes it is a `byte[]`. The difference seems quite important to me.